### PR TITLE
[iOS] Rewrite the guide to integrate with Brownfield apps

### PR DESCRIPTION
**Pull request by cipolleschi**
The page Integrate with an Existing app is outdated and it contains wrong information
Following the recent work on Android, I revamped the iOS part as well.

I also unified the Objective-C and Swift part given that there are not many differences today to justify having two separate pages.

### CODE REVIEW

File: docs/_integration-with-existing-apps-ios.md [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+<TabItem value="yarn">
+
+```shell
+yarn install
+```
+
+</TabItem>
+</Tabs>
+
+Installation process has created a new `/node_modules` folder. This folder stores all the JavaScript dependencies required to build your project.
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:56]:
```suggestion
Installation process has created a new `node_modules` folder. This folder stores all the JavaScript dependencies required to build your project.
```

File: docs/_integration-with-existing-apps-ios.md [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+
+Install the Command Line Tools. Choose **Settings... (or Preferences...)** in the Xcode menu. Go to the Locations panel and install the tools by selecting the most recent version in the Command Line Tools dropdown.
+
+![Xcode Command Line Tools](/docs/assets/GettingStartedXcodeCommandLineTools.png)
+
+### CocoaPods
+
+[CocoaPods](https://cocoapods.org) is a package management tool for iOS and macOS development. We use it to add the actual React Native framework code locally into your current project.
+
+We recommend installing CocoaPods using [Homebrew](https://brew.sh/).
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:72]:
```suggestion
We recommend installing CocoaPods using [Homebrew](https://brew.sh/):
```

File: docs/_integration-with-existing-apps-ios.md [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+
+```shell
+brew install cocoapods
+```
+
+> It is technically possible not to use CocoaPods, but that would require manual library and linker additions that would overly complicate this process.
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:78]: This adds nothing to this page. I would just remove it

File: docs/_integration-with-existing-apps-ios.md [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+```sh
+wget https://raw.githubusercontent.com/react-native-community/template/refs/heads/0.76-stable/template/ios/Podfile
+```
+
+Please use the Community Template as a reference point for the [Gemfile](https://github.com/react-native-community/template/blob/0.76-stable/template/Gemfile) and for the [Podfile](https://github.com/react-native-community/template/blob/0.76-stable/template/ios/Podfile).
+
+:::note
+Remember to change [this line](https://github.com/react-native-community/template/blob/0.76-stable/template/ios/Podfile#L17) and [this line](https://github.com/react-native-community/template/blob/0.76-stable/template/ios/Podfile#L26) of the Podfile to match the name of your app.
+
+If your app don't have tests, rememebr to remove [this block](https://github.com/react-native-community/template/blob/0.76-stable/template/ios/Podfile#L26-L29).
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:107]:
```suggestion
If your app don't have tests, remember to remove [this block](https://github.com/react-native-community/template/blob/0.76-stable/template/ios/Podfile#L26-L29).
```

File: docs/_integration-with-existing-apps-ios.md [hunk 5]
```diff
@@ -0,0 +1,598 @@
...
+
+Here the [Community template file as reference](https://github.com/react-native-community/template/blob/0.76-stable/template/App.tsx)
+
+## 5. Integrating with your iOS code
+
+We now need to add some native code in order to start the React Native runtime and tell it to render our React components.
+
+### Requirements
+
+React Native is supposed to work with the AppDelegate. Sadly it does not supports the SceneDelegate pattern as of today, but we are getting closer to it.
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:224]:
```suggestion
React Native is supposed to work with the `AppDelegate`. Sadly it does not supports the `SceneDelegate` pattern as of today, but we are getting closer to it.
```

File: docs/_integration-with-existing-apps-ios.md [hunk 6]
```diff
@@ -0,0 +1,598 @@
...
+```
+
+Here the [Community template file as reference](https://github.com/react-native-community/template/blob/0.76-stable/template/App.tsx)
+
+## 5. Integrating with your iOS code
+
+We now need to add some native code in order to start the React Native runtime and tell it to render our React components.
+
+### Requirements
+
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:223]: > but we are getting closer to it

If you have a GH issue to link, please add it. Otherwise I would remove this reference

File: docs/_integration-with-existing-apps-ios.md [hunk 7]
```diff
@@ -0,0 +1,598 @@
...
+React Native is supposed to work with the AppDelegate. Sadly it does not supports the SceneDelegate pattern as of today, but we are getting closer to it.
+
+The following part assumes that you don't have a SceneDelegate and that your application AppDelegate looks like this:
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:226]:
```suggestion
The following part assumes that your `AppDelegate` looks like this:
```

File: docs/_integration-with-existing-apps-ios.md [hunk 8]
```diff
@@ -0,0 +1,598 @@
...
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Update the AppDelegate class
+
+First, we need to extends the AppDelegate to inherit from one of the classes provided by React Native: `RCTAppDelegate`.
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:279]:
```suggestion
First, we need to extends the `AppDelegate` to inherit from one of the classes provided by React Native: `RCTAppDelegate`.
```

File: docs/_integration-with-existing-apps-ios.md [hunk 9]
```diff
@@ -0,0 +1,598 @@
...
+#import "ViewController.h"
++#import "ReactViewController.h"
+
+@interface ViewController ()
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
```
Comment (cortinico) [inline docs/_integration-with-existing-apps-ios.md:420]: Is this correct with the `-`?

**Review by cortinico — APPROVED** [review]
Great work! Mostly minor nits

File: docs/_integration-with-existing-apps-ios.md (hunk 9 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:420]: yes, in objc this denotes an instance method. a `+` instead, denotes a static method. Yes, the language is weird.

File: docs/_integration-with-existing-apps-ios.md (hunk 1 above)
Comment (cortinico) [reply docs/_integration-with-existing-apps-ios.md:56]: Yes please 🙏

File: docs/_integration-with-existing-apps-ios.md [hunk 10]
```diff
@@ -0,0 +1,596 @@
...
+1. Set up the correct directory structure.
+2. Install the necessary NPM dependencies.
+3. Adding React Native to your Podfile configuration.
+4. Writing the TypeScript code for your first React Native screen.
+5. Integrate React Native with your iOS code using a `RCTRootView`.
+6. Testing your integration by running the bundler and seeing your app in action.
+
+## Using the Community Template
+
+While you follow this guide, we suggest you to use the [React Native Community Template](https://github.com/react-native-community/template/) as reference. The template contains a **minimal Android app** and will help you understanding how to integrate React Native into an existing Android app.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:16]:
```suggestion
While you follow this guide, we suggest you to use the [React Native Community Template](https://github.com/react-native-community/template/) as reference. The template contains a **minimal iOS app** and will help you understanding how to integrate React Native into an existing iOS app.
```

File: docs/_integration-with-existing-apps-ios.md (hunk 5 above)
Comment (blakef) [reply docs/_integration-with-existing-apps-ios.md]: Can we link to docs for AppDelegate and SceneDelegate?

File: docs/_integration-with-existing-apps-ios.md [hunk 11]
```diff
@@ -0,0 +1,596 @@
...
+```shell
+yarn install
+```
+
+</TabItem>
+</Tabs>
+
+Installation process has created a new `node_modules` folder. This folder stores all the JavaScript dependencies required to build your project.
+
+Add `node_modules/` to your `.gitignore` file (here the [Community default one](https://github.com/react-native-community/template/blob/0.76-stable/template/_gitignore)).
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:58]: nit: We can also suggest to use wget here

File: docs/_integration-with-existing-apps-ios.md [hunk 12]
```diff
@@ -0,0 +1,596 @@
...
+
+### Command Line Tools for Xcode
+
+Install the Command Line Tools. Choose **Settings... (or Preferences...)** in the Xcode menu. Go to the Locations panel and install the tools by selecting the most recent version in the Command Line Tools dropdown.
+
+![Xcode Command Line Tools](/docs/assets/GettingStartedXcodeCommandLineTools.png)
+
+### CocoaPods
+
+[CocoaPods](https://cocoapods.org) is a package management tool for iOS and macOS development. We use it to add the actual React Native framework code locally into your current project.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:70]: I'm not sure about this sentence:

> We use it to add the actual React Native framework code locally into your current project.

Maybe something like: We use it to integrate React Native's code and dependencies into your current project.

File: docs/_integration-with-existing-apps-ios.md [hunk 13]
```diff
@@ -0,0 +1,598 @@
...
++    let factory = (RCTSharedApplication()?.delegate as? RCTAppDelegate)?.rootViewFactory;
++    self.view = factory?.view(withModuleName: "HelloWorld");
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+4. Make sure to disable the Sandbox scripting. To achieve this, in Xcode, click on your app, then on build settings. Filter for script and set the `User Script Sandboxing` to `NO`.
```
Comment (blakef) [inline docs/_integration-with-existing-apps-ios.md:529]: Can you add a sentence about why you disable Sandbox scripting?  If it isn't important please ignore, just the first question that popped into my head.

File: docs/_integration-with-existing-apps-ios.md [hunk 14]
```diff
@@ -0,0 +1,596 @@
...
+To configure CocoaPods, we need two files:
+
+- A **Gemfile** that defines which Ruby dependencies we need.
+- A **Podfile** that defines how to properly install our dependencies.
+
+For the **Gemfile**, go to the root directory of your project and run this command
+
+```sh
+wget https://raw.githubusercontent.com/react-native-community/template/refs/heads/0.76-stable/template/Gemfile
+```
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:91]: This could be marked as optional

File: docs/_integration-with-existing-apps-ios.md [hunk 15]
```diff
@@ -0,0 +1,596 @@
...
+```js
+import {AppRegistry} from 'react-native';
+import App from './App';
+
+AppRegistry.registerComponent('HelloWorld', () => App);
+```
+
+### Create a `App.tsx` file
+
+Then, let's create a `App.tsx` file. This is a TypeScript file that will contain the React Native component that we will integrate into our Android application.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:141]:
```suggestion
Then, let's create a `App.tsx` file. This is a TypeScript file that will contain the React Native component that we will integrate into our iOS application.
```

File: docs/_integration-with-existing-apps-ios.md [hunk 16]
```diff
@@ -0,0 +1,596 @@
...
+
+Here the [Community template file as reference](https://github.com/react-native-community/template/blob/0.76-stable/template/App.tsx)
+
+## 5. Integrating with your iOS code
+
+We now need to add some native code in order to start the React Native runtime and tell it to render our React components.
+
+### Requirements
+
+React Native is supposed to work with the `AppDelegate`. Sadly it does not supports the `SceneDelegate` pattern as of today.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:222]: This sets a negative tone, I would rephrase this to: 

React Native integrates into your project by subclassing `RCTAppDelegate`. Currently `SceneDelegate` is not supported.

or something like this, hopefully, we will remove this soon 🤞

Comment (okwasniewski) [reply docs/_integration-with-existing-apps-ios.md:222]: Also if the user wants he can pass `automaticallyLoadReactNativeWindow = NO;` and then init his view in SceneDelegate by referencing RCTRootViewFactory from AppDelegate so its not like SceneDelegate won't work at all.

File: docs/_integration-with-existing-apps-ios.md [hunk 17]
```diff
@@ -0,0 +1,596 @@
...
+### 1. Set up directory structure
+
+To ensure a smooth experience, create a new folder for your integrated React Native project, then **move your existing iOS project** to the `/ios` subfolder.
+
+## 2. Install NPM dependencies
+
+Go to the root directory and run the following command:
+
+```
+wget https://raw.githubusercontent.com/react-native-community/template/refs/heads/0.76-stable/template/package.json
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:32]: Is it possible to move this `0.76-stable` into a constant? This could make future maintenance easier

File: docs/_integration-with-existing-apps-ios.md [hunk 18]
```diff
@@ -0,0 +1,596 @@
...
+
+Once you reach your React-powered Activity inside the app, it should load the JavaScript code from the development server and display:
+
+![Screenshot](/docs/assets/EmbeddedAppIOSVideo.gif)
+
+### Creating a release build in Android Studio
+
+You can use Xcode to create your release builds too! The only additional step is to add a script that is executed when the app is built to package your JS and images into the iOS application.
+
+1. In Xocde, select your application
```
Comment (blakef) [inline docs/_integration-with-existing-apps-ios.md:575]:
```suggestion
1. In Xcode, select your application
```

File: docs/_integration-with-existing-apps-ios.md [hunk 19]
```diff
@@ -0,0 +1,596 @@
...
++#else
++  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
++#endif
++}
+ @end
+```
+
+Let's have a look at the code above:
+
+1. We are inherithing from the `RCTAppDelegate` and we are calling the `applicatinon:didFinishLaunchingWithOptions` of the `RCTAppDelegate`. This delegates all the React Native initialization processes to the base class.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:340]:
```suggestion
1. We are inheriting from the `RCTAppDelegate` and we are calling the `application:didFinishLaunchingWithOptions` of the `RCTAppDelegate`. This delegates all the React Native initialization processes to the base class.
```

File: docs/_integration-with-existing-apps-ios.md [hunk 20]
```diff
@@ -0,0 +1,596 @@
...
+2. We are customizing the RCTAppDelegate by setting the `automaticallyLoadReactNativeWindow` to `NO`. This step instruct React Native that the app is handling the `UIWindow` and React Native should not worry about that.
+3. The methods `sourceURLForBridge:` and `bundleURL` are used by the App to tell to React Native wher it can find the JS bundle that needs to be rendered. The `sourceURLForBridge:` is from the Old Architecture and you can see that it is deferring the decision to the `bundleURL` method, required by the New Architecture.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:342]:
```suggestion
3. The methods `sourceURLForBridge:` and `bundleURL` are used by the App to tell React Native where it can find the JS bundle that needs to be rendered. The `sourceURLForBridge:` is from the Old Architecture and you can see that it is deferring the decision to the `bundleURL` method, required by the New Architecture.
```

File: docs/_integration-with-existing-apps-ios.md [hunk 21]
```diff
@@ -0,0 +1,596 @@
...
++    window.makeKeyAndVisible()
+    return true
+  }
+
++  override func sourceURL(for bridge: RCTBridge) -> URL? {
++    self.bundleURL()
++  }
+
++  override func bundleURL() -> URL? {
++#if DEBUG
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:380]: It might be worth mentioning that users should check Active Compilation Conditions. In theory this should be set but if its not then this may lead to some issues 

image

File: docs/_integration-with-existing-apps-ios.md [hunk 22]
```diff
@@ -0,0 +1,596 @@
...
++#else
++    Bundle.main.url(forResource: "main", withExtension: "jsbundle")
++#endif
++  }
+}
+```
+
+Let's have a look at the code above:
+
+1. We are inherithing from the `RCTAppDelegate` and we are calling the `applicatinon(_:didFinishLaunchingWithOptions:)` of the `RCTAppDelegate`. This delegates all the React Native initialization processes to the base class.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:391]:
```suggestion
1. We are inheriting from the `RCTAppDelegate` and we are calling the `application (_:didFinishLaunchingWithOptions:)` of the `RCTAppDelegate`. This delegates all the React Native initialization processes to the base class.
```

File: docs/_integration-with-existing-apps-ios.md [hunk 23]
```diff
@@ -0,0 +1,598 @@
...
+bundle exec pod install
+```
+
+The first command will install the ruby dependencies and the second command will actually integrate the React Native code in your application so that your iOS files can import the React Native headers.
+
+## 5. Writing the TypeScript Code
+
+Now we will actually modify the native Android application to integrate React Native.
+
+The first bit of code we will write is the actual React Native code for the new screen that will be integrated into our application.
```
[the remark this answers is not in the document]
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:122]: My personal interpretation is that a screen is the whole screen. A surface could be just a view or a set of views that occupy part of the screen.

File: docs/_integration-with-existing-apps-ios.md [hunk 24]
```diff
@@ -0,0 +1,598 @@
...
+
+To configure CocoaPods, we need two files:
+
+- A **Gemfile** that defines which Ruby dependencies we need.
+- A **Podfile** that defines how to properly install our dependencies.
+
+For the **Gemfile**, go to the root directory of your project and run this command
+
+```sh
+wget https://raw.githubusercontent.com/react-native-community/template/refs/heads/0.76-stable/template/Gemfile
```
[the remark this answers is not in the document]
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:90]: Q: should we update the Android guide too then?

File: docs/_integration-with-existing-apps-ios.md [hunk 25]
```diff
@@ -0,0 +1,596 @@
...
+
+@end
+
+@implementation ViewController
+
+ - (void)viewDidLoad {
+   [super viewDidLoad];
+   // Do any additional setup after loading the view.
+   self.view.backgroundColor = UIColor.systemBackgroundColor;
++  UIButton * button = [UIButton new];
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:422]:
```suggestion
+  UIButton *button = [UIButton new];
```

File: docs/_integration-with-existing-apps-ios.md (hunk 13 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:529]: Sure, it's for Hermes, anyway.

File: docs/_integration-with-existing-apps-ios.md [hunk 26]
```diff
@@ -0,0 +1,596 @@
...
+</TabItem>
+</Tabs>
+
+4. Make sure to disable the Sandbox scripting. To achieve this, in Xcode, click on your app, then on build settings. Filter for script and set the `User Script Sandboxing` to `NO`.
+
+![Disable Sandboxing](/docs/assets/disable-sandboxing.png);
+
+## 6. Test your integration
+
+You have now done all the basic steps to integrate React Native with your current application. Now we will start the [Metro bundler](https://metrobundler.dev/) to build the TypeScript code of your application and have the server running on localhost to serve it.
```
[the remark this answers is not in the document]
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:535]: we should update this in Android too.

File: docs/_integration-with-existing-apps-ios.md [hunk 27]
```diff
@@ -0,0 +1,596 @@
...
++    [button.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
++    [button.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
++    [button.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor],
++    [button.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
++  ]];
+ }
+
++- (void)presentReactNative
++{
++  [self presentViewController:[ReactViewController new] animated:YES completion:nil];
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:440]: Shouldn't we retain this ViewController as a property on the class?

I am not sure but I think I've run into some memory management issues with deallocation when doing it this way

File: docs/_integration-with-existing-apps-ios.md [hunk 28]
```diff
@@ -0,0 +1,596 @@
...
+@interface ReactViewController ()
+
+@end
+
+@implementation ReactViewController
+
+ - (void)viewDidLoad {
+   [super viewDidLoad];
+   // Do any additional setup after loading the view.
++   RCTRootViewFactory * factory = ((RCTAppDelegate *)RCTSharedApplication().delegate).rootViewFactory;
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:501]:
```suggestion
+   RCTRootViewFactory *factory = ((RCTAppDelegate *)RCTSharedApplication().delegate).rootViewFactory;
```

File: docs/_integration-with-existing-apps-ios.md [hunk 29]
```diff
@@ -0,0 +1,596 @@
...
+import UIKit
++import React_RCTAppDelegate
+
+class ReactViewController: UIViewController {
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
++    let factory = (RCTSharedApplication()?.delegate as? RCTAppDelegate)?.rootViewFactory;
++    self.view = factory?.view(withModuleName: "HelloWorld");
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:521]:
```suggestion
+    let factory = (RCTSharedApplication()?.delegate as? RCTAppDelegate)?.rootViewFactory
+    self.view = factory?.view(withModuleName: "HelloWorld")
```

File: docs/_integration-with-existing-apps-ios.md [hunk 30]
```diff
@@ -0,0 +1,596 @@
...
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+4. Make sure to disable the Sandbox scripting. To achieve this, in Xcode, click on your app, then on build settings. Filter for script and set the `User Script Sandboxing` to `NO`.
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:529]: Might be worth explaining why this is needed

File: docs/_integration-with-existing-apps-ios.md [hunk 31]
```diff
@@ -0,0 +1,596 @@
...
+</TabItem>
+</Tabs>
+
+Now build and run your Android app as normal.
+
+Once you reach your React-powered Activity inside the app, it should load the JavaScript code from the development server and display:
+
+![Screenshot](/docs/assets/EmbeddedAppIOSVideo.gif)
+
+### Creating a release build in Android Studio
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:571]:
```suggestion
### Creating a release build in Xcode
```

File: website/core/TabsConstants.js [hunk 32]
```diff
@@ -32,6 +32,12 @@ const androidLanguages = [
 ];
 const defaultAndroidLanguage = 'java';
 
+const appleLanguages = [
+  {label: 'ObjectiveC', value: 'objc'},
+  {label: 'Swift', value: 'swift'},
+];
+const defaultAppleLanguage = 'objc';
```
Comment (okwasniewski) [inline website/core/TabsConstants.js:39]: How about swift as the default? 😆

**Review by okwasniewski — COMMENTED** [review]
Great work overall! Added few comments

**Review by okwasniewski — COMMENTED** [review]
Great work overall! Added few comments

File: docs/_integration-with-existing-apps-ios.md (hunk 11 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:58]: I thought about it, but they might not be using git as source control tool.

File: docs/_integration-with-existing-apps-ios.md (hunk 14 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:91]: Yes, but there are som many issues that can occur if you don't use the right gems that I'd rather keep it.

File: docs/_integration-with-existing-apps-ios.md (hunk 16 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:222]: Interesting... let's remove this for now. We can integrate the `SceneDelegate` in a next iteration

File: docs/_integration-with-existing-apps-ios.md (hunk 17 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:32]: no idea, honestly.... We should look into it.

File: docs/_integration-with-existing-apps-ios.md (hunk 21 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:380]: Debug and Release flavors are created by default by Xcode. The only case when that flag is not there is for a project that started as Objective-C and got migrated to Swift - so Xcode didn't add that automation. 
But that's an edge case that people working on migrated projects should have already handled.
I'd rather keep the guide minimal, if possible.

File: docs/_integration-with-existing-apps-ios.md (hunk 25 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:422]: we have a linter that fixes the code in the markdown in the website. :wink:

File: docs/_integration-with-existing-apps-ios.md (hunk 27 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:440]: It depends on what the VC does. In this simple example it should not be required, but let add it for safety.

File: website/core/TabsConstants.js (hunk 32 above)
Comment (cipolleschi) [reply website/core/TabsConstants.js:39]: Too early! XD

File: docs/_integration-with-existing-apps-ios.md [hunk 33]
```diff
@@ -0,0 +1,605 @@
...
+```
+
+</TabItem>
+</Tabs>
+
+Now build and run your iOS app as normal.
+
+Once you reach your React-powered Activity inside the app, it should load the JavaScript code from the development server and display:
+
+![Screenshot](/docs/assets/EmbeddedAppIOSVideo.gif)
```
Comment (blakef) [inline docs/_integration-with-existing-apps-ios.md:578]: Can we constrain the width (see [example code here](https://github.com/facebook/react-native-website/blob/61313ad2e9f756ea0c3e6dd87a1fd53e1f674d6e/docs/datepickerios.md?plain=1#L90))?

Currently it's like this:
CleanShot 2024-10-02 at 12 05 52@2x

If I set a max with of 300px it looks like this:
CleanShot 2024-10-02 at 12 06 28@2x

File: docs/_integration-with-existing-apps-ios.md [hunk 34]
```diff
@@ -0,0 +1,605 @@
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
+
+## Key Concepts
+
+The keys to integrating React Native components into your iOS application are to:
```
Comment (okwasniewski) [inline docs/_integration-with-existing-apps-ios.md:5]: Maybe we can also mention that this guide covers both old and new architecture? And if you want to use new arch you can install pods with a flag?

Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:5]: this will be new arch only. All the references are to 0.76 and it will be published with that.
It follows the usual guidelines to disable it, eventually. But this is out of the scope of this guide, imho.

File: docs/_integration-with-existing-apps-ios.md (hunk 33 above)
Comment (cipolleschi) [reply docs/_integration-with-existing-apps-ios.md:578]: we need to do it on android as well.
